pageserver/src/tenant/remote_timeline_client.rs

```diff
@@ -501,14 +501,20 @@ impl RemoteTimelineClient {
 
     /// Download a (layer) file from `path`, into local filesystem.
     ///
-    /// 'layer_metadata' is the metadata from the remote index file.
-    ///
     /// On success, returns the size of the downloaded file.
     pub async fn download_layer_file(
         &self,
         layer_file_name: &LayerFileName,
-        layer_metadata: &LayerFileMetadata,
     ) -> anyhow::Result<u64> {
+        let layer_metadata = {
+            let q = self.upload_queue.lock().unwrap();
+            q.get_layer_metadata(layer_file_name)?.ok_or_else(|| {
+                anyhow::anyhow!(
+                    "Tried to download layer file that does not exist in remote metadata"
+                )
+            })?
+        };
```

Comment:
Technically, we need a `wait_completion()` before calling `get_layer_metadata`.
Reason: we update `latest_files` early, when we schedule the upload, not when the upload is completed.
I guess it doesn't matter in practice because we should never be asking to download a layer that hasn't been uploaded yet.

Reply:
> I guess it doesn't matter in practice because we should never be asking to download a layer that hasn't been uploaded yet.

Right, that's the intent.